### ISSUE TITLE
Standardised `Nullable<T>` type

### ISSUE THREAD

**Issue by J-Cake** [microsoft/TypeScript#49984]
# Suggestion

### Primary Suggestion
I'm not sure why this hasn't been standardised yet, I'm sure there's a good explanation, but basically, the idea is that
```typescript
type Nullable<T> = T | null | undefined;
```
be introduced into the standard list of types. 

###
Secondary Suggestion
* I also happened to notice that variables cannot be marked with `?` directly. My suggestion would be that 
```typescript
let x?: number;
```
translates to 
```typescript
let x: Nullable<number>;
```
List of keywords you searched for before creating this issue. Write them down here so that others can find this suggestion more easily and help provide feedback.

## ✅ Viability Checklist

My suggestion meets these guidelines:

* [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, new syntax sugar for JS, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

## ⭐ Suggestion

Basically I want to be able to do:
```typescript
const x?: Date = Api?.getTime();
```

* It's much cleaner
* I use exactly that type in **every single project**
* I tend to redefine it in most of my files instead of importing it because I can't think of a sensible place to put it
* me want :P

**Comment by MartinJohns** [comment]
Duplicate of #39522 and many others. Used search terms: `nullable in:title`

**Comment by MartinJohns** [comment]
Your secondary suggestion is a duplicate of #41901. Used search terms: `optional variable in:title`
